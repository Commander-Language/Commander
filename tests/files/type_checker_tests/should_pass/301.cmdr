//Test builtin ln type checks
a = 1;
x = ln(16);
x = ln(0.25);
x = ln((1.0, true, 3)[a]);
