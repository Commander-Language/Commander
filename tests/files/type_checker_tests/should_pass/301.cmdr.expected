(PrgmNode
	(ExprStmtNode (BinOpExprNode (BindingNode a) SET (IntExprNode 1 INT) INT))
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (CallExprNode (LValueExprNode (VarExprNode ln (fn (INT) -> FLOAT)) (fn (INT) -> FLOAT)) (ExprsNode (IntExprNode 16 INT)) FLOAT) FLOAT))
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (CallExprNode (LValueExprNode (VarExprNode ln (fn (FLOAT) -> FLOAT)) (fn (FLOAT) -> FLOAT)) (ExprsNode (FloatExprNode 0.250000 FLOAT)) FLOAT) FLOAT))
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (CallExprNode (LValueExprNode (VarExprNode ln (fn (INT) -> FLOAT)) (fn (INT) -> FLOAT)) (ExprsNode (IndexExprNode (TupleExprNode (ExprsNode (FloatExprNode 1.000000 FLOAT) (BoolExprNode true BOOL) (IntExprNode 3 INT)) (FLOAT, BOOL, INT)) (LValueExprNode (VarExprNode a INT) INT))) FLOAT) FLOAT)))
